concent_api/common/tests/test_api_view.py

```diff
@@ -218,10 +219,23 @@ def setUp(self):
         message_timestamp = get_current_utc_timestamp() + deadline_offset
         compute_task_def = ComputeTaskDefFactory()
         compute_task_def['deadline'] = message_timestamp
+        want_to_compute_task = WantToComputeTaskFactory(
+            provider_public_key=encode_hex(PROVIDER_PUBLIC_KEY),
+        )
+        want_to_compute_task = load(
+            dump(
+                want_to_compute_task,
+                PROVIDER_PRIVATE_KEY,
+                settings.CONCENT_PUBLIC_KEY,
+            ),
+            settings.CONCENT_PRIVATE_KEY,
+            PROVIDER_PUBLIC_KEY,
+            check_time=False,
+        )
         task_to_compute = tasks.TaskToComputeFactory(
             compute_task_def=compute_task_def,
             requestor_public_key=encode_hex(REQUESTOR_PUBLIC_KEY),
-            provider_public_key=encode_hex(PROVIDER_PUBLIC_KEY),
+            want_to_compute_task=want_to_compute_task,
```

Comment:
`TaskToComputeFactory` have possibility to create nested `WantToComputeTask` with custom key like this:
```
task_to_compute = tasks.TaskToComputeFactory(
    ...
    want_to_compute_task__provider_public_key=encode_hex(PROVIDER_PUBLIC_KEY),
)
```

There is no need to create `WantToComputeTask` manually.

Reply:
There is a method in `TaskToCompute` which adds signature to `TaskToCompute`, so the nested messages inside it should be signed along. If not, keep the current code but report issue to golem-messages.

Reply:
I don't think so, because `TaskToCompute` is sign by requestor, where `WantToComputeTask` must be sign by provider.